### PR TITLE
Add go modules

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
Use go modules to build che-machine-exec

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #
 # Dockerfile defines che-machine-exec production image eclipse/che-machine-exec-dev
 #
-FROM golang:1.10.3-alpine as go_builder
+FROM golang:1.12.8-alpine as go_builder
```
Comment (sleshchenko) [inline Dockerfile]: 1. Did you consider option where we remove vendor folder?
2. If you want to keep vendor folder, should not we add `-mod=vendor` to go build args? See https://github.com/golang/go/wiki/Modules#how-do-i-use-vendoring-with-modules-is-vendoring-going-away
3. Consider simplifying Dockerfile a bit, like `WORKDIR /go/src/github.com/eclipse/che-machine-exec/` now can be just `WORKDIR /che-machine-exec/`

File: vendor/modules.txt [hunk 2]
```diff
@@ -0,0 +1,213 @@
+# github.com/davecgh/go-spew v1.1.0
```
Comment (sleshchenko) [inline vendor/modules.txt]: Is vendor folder in actual state? Seems no because I run `go mod vendor` and vendor folder is changed. Please recheck it

Comment (AndrienkoAleksandr) [reply vendor/modules.txt]: Done. Finally Fixed.

**Comment by AndrienkoAleksandr** [comment]
@sleshchenko Take a look, please.

**Comment by AndrienkoAleksandr** [comment]
> Just one comment - vendor folder seems to have an outdated state. I checked it with go 1.12.8 + 1.13.6. 

Hello, thanks, I reproduced the same with go go1.14.1. But initially I did pr in the development Che workspace with golang go1.12.16 using devfile. Looks like newer go has some improvements to detect unused files. Looks like we need newer golang for devfile. I will care about that in the next pr's.

**Pull request merged**